### PR TITLE
[RB] - only load stripe SDK on intent

### PR DESCRIPTION
**Pull request by rBangay**
## What does this change?
only load the stripe sdk when the user explicitly decides to update their payment details:

Screenshot 2020-08-24 at 10 36 28

I've taken some screenshots in order to explain the approach here ...

The first image is the default route (account overview) on the existing master branch
01_master

The 2nd image is if we move the loadStripe call within the useEffect hook but keep the existing stripe imports (again on the default route - account overview):
02_loadstripeInUseEffect

The 3rd and 4th images are the approach taken in this pr (the 1st image being the default route and the 2nd being the update payment page after capthca is complete)
03_asyncLoadStripeImport

update payment page:
04_asyncLoadStripeImport

### CODE REVIEW

File: app/client/components/payment/update/card/cardInputForm.tsx [hunk 1]
```diff
@@ -64,7 +67,17 @@ export const CardInputForm = (props: CardInputFormProps) => {
...
+    import("@stripe/stripe-js").then(StripeAsync => {
+      setStripePromise(StripeAsync.loadStripe(props.stripeApiKey));
+    });
+  };
+
   const loadSetupIntent = (recaptchaToken: string) => {
+    if (stripePromise === null) {
+      loadStripe();
+    }
+
```
Comment (twrichards) [inline app/client/components/payment/update/card/cardInputForm.tsx:80]: this is not synchronous? think you might have a race condition?

File: app/client/components/payment/update/card/cardInputForm.tsx [hunk 2]
```diff
@@ -38,7 +38,10 @@ export const CardInputForm = (props: CardInputFormProps) => {
   >();
   const [stripeSetupIntentError, setstripeSetupIntentError] = useState<Error>();
 
-  const stripePromise = loadStripe(props.stripeApiKey);
```
Comment (twrichards) [inline app/client/components/payment/update/card/cardInputForm.tsx:41]: could this not have been moved into a `useEffect` instead?

**Review by twrichards — COMMENTED** [review]
Is this approach followed from somewhere?

File: app/client/components/payment/update/card/cardInputForm.tsx (hunk 1 above)
Comment (rBangay) [reply app/client/components/payment/update/card/cardInputForm.tsx:80]: > this is not synchronous? think you might have a race condition?

I've taken some screenshots in order to explain the approach here ...

The first image is the default route (account overview) on the existing master branch
01_master


The 2nd image is if we move the loadStripe call within the useEffect hook but keep the existing stripe imports (again on the default route - account overview):
02_loadstripeInUseEffect

The 3rd and 4th images are the approach taken in this pr (the 1st image being the default route and the 2nd being the update payment page after capthca is complete)
03_asyncLoadStripeImport

update payment page:
04_asyncLoadStripeImport

**Review by twrichards — APPROVED** [review]
Great work! Thanks for talking it through with me IRL. LGTM 👍 

@paulbrown1982 will be pleased.

File: app/client/components/payment/update/card/cardInputForm.tsx (hunk 2 above)
Comment (twrichards) [reply app/client/components/payment/update/card/cardInputForm.tsx:41]: thanks for explaining in the other answer :)

**Pull request merged**